### PR TITLE
update_deps.sh prior to release-1.18 branch cut

### PR DESCRIPTION
**Pull request by pmerrison**
**Please provide a description of this PR:**
Output of `UPDATE_BRANCH=master ./bin/update_deps.sh; make gen` command

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -88,8 +88,8 @@ require (
 	gopkg.in/yaml.v2 v2.4.0
 	gopkg.in/yaml.v3 v3.0.1
 	helm.sh/helm/v3 v3.11.2
-	istio.io/api v0.0.0-20230411172800-cb047abbdf91
-	istio.io/client-go v1.18.0-alpha.0.0.20230411173201-f27e09ad6e63
+	istio.io/api v0.0.0-20230406193325-c74882c8130f
```
Comment (zirain) [inline go.mod:91]: revert?

Comment (ericvn) [reply go.mod:91]: As noted in the channel, one needs to be extra vigilant with update_deps or any job which is simply doing a `go get`. As noted here, you can see the data is moving back which is indicative of an issue.

**Review by ericvn — CHANGES_REQUESTED** [review]

**Comment by ericvn** [comment]
As a sanity check:
```
> grep istio.io go.mod
module istio.io/istio
	istio.io/api v0.0.0-20230411172800-cb047abbdf91
	istio.io/client-go v1.18.0-alpha.0.0.20230411173201-f27e09ad6e63
	istio.io/pkg v0.0.0-20230411034200-2c98fd007de2
```
And looking at the latest commit in the 3 repos, they match, so no go updates are needed.

File: prow/release-commit.sh [hunk 2]
```diff
@@ -36,7 +36,7 @@ DOCKER_HUB=${DOCKER_HUB:-gcr.io/istio-testing}
 GCS_BUCKET=${GCS_BUCKET:-istio-build/dev}
 
 # Use a pinned version in case breaking changes are needed
-BUILDER_SHA=49103b5e8fa57db9167608e529fb6c9080427d70
+BUILDER_SHA=b2ac5974979f2b24771dc6412bff514c282891d1
```
Comment (ericvn) [inline prow/release-commit.sh:39]: Looking at the latest commit in this repo, which we get differently, so not affect by `go get`, we see that this update is correct.

**Comment by ericvn** [comment]
So if you can revert the go.* file changes in this PR, I think it would be ready to go.

**Review by ericvn — APPROVED** [review]
Looks good now.

**Pull request closed without merging**